pytorch_lightning/core/optimizer.py

```diff
@@ -121,6 +121,16 @@ def toggle_model(self, sync_grad: bool = True):
             yield
             self._untoggle_model()
 
+    @contextmanager
+    def __fault_tolerant_supported(self) -> Generator:
+        """The closure should prevent execution fault tolerance within the parameters update.
+
+        Restore the fault tolerant state accordingly
```

Comment:
Is this asuming that the inner code (the one that yields) will change the value of `self._trainer._fault_tolerant_possible`?

Is the purpose here to set the previous value in case that happens?

This is very ugly in terms of side effects and misdirection.

Reply:
Yes, this was a solution to complete the `self._trainer._fault_tolerant_possible=False` applied after the closure is being run.